feat(healthcheck): return JSON with status info

Abo-Omar-74 opened the pull request:
- update healthcheck handler to return status, environment, version, servers, ready
- update unit test for new JSON response

aaronbrethorst reviewed (changes requested):
great work! there are two changes I'd like to see before merging:

1. Rebase on top of the latest bits in `main`
2. Incorporate the copilot feedback

Abo-Omar-74 commented:
I have made the required changes:
- Added failure test case for healthcheckHandler (HTTP 500 scenario)
- Updated the documentation of `healthcheckHandler` function 
- Rebased onto latest main
